dissect/target/containers/vhd.py

```diff
@@ -18,12 +18,14 @@ def __init__(self, fh: Union[BinaryIO, Path], *args, **kwargs):
 
     @staticmethod
     def detect_fh(fh: BinaryIO, original: Union[list, BinaryIO]) -> bool:
-        offset = fh.tell()
-        fh.seek(-512, io.SEEK_END)
-        magic = fh.read(9)
-        fh.seek(offset)
-
-        return b"conectix" in magic
+        try:
+            offset = fh.tell()
+            fh.seek(-512, io.SEEK_END)
+            magic = fh.read(9)
+            fh.seek(offset)
+            return b"conectix" in magic
+        except Exception:
+            return False
```

Comment:
JFFS2 filesystems can be [smaller than 512 bytes](https://github.com/fox-it/dissect.jffs/pull/1/files#diff-b86fab974720cbe31570915ef800d0a155a2bf807fd3bbca378855bdf93d0a43). Please let us know if a different fix in `vhd.py` is desired.

Reply:
The volume and filesystem detect methods have a small wrapper that deals with errors like these. We'll need to add that to container down the line too. For now, this will do. Could you change the `Exception` to the specific error that is raised (`OSError` IIRC)?

Reply:
Added in 4729cde6cdfde5a5368f1adbad6daa71d9fa9732